modules/tinymce/src/plugins/fullscreen/test/ts/webdriver/FullscreenTrapFocusTest.ts

```diff
@@ -0,0 +1,143 @@
+import { FocusTools, RealKeys, Waiter } from '@ephox/agar';
+import { context, describe, it } from '@ephox/bedrock-client';
+import { Arr, Optional } from '@ephox/katamari';
+import { Attribute, Insert, Remove, SugarDocument, SugarElement } from '@ephox/sugar';
+import { McEditor, TinyContentActions, TinyDom, TinyHooks } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+import * as NativeFullscreen from 'tinymce/plugins/fullscreen/core/NativeFullscreen';
+
+import FullscreenPlugin from '../../../main/ts/Plugin';
+
+describe('webdriver.tinymce.plugins.fullscreen.FullscreenTrapFocusTest', () => {
+  const pDoShiftTab = async (selector: string = 'iframe => body'): Promise<void> => {
+    await RealKeys.pSendKeysOn(selector, [ RealKeys.combo({ shift: true }, 'tab') ]);
+  };
+
+  const pDoTab = async (selector: string = 'iframe => body'): Promise<void> => {
+    await RealKeys.pSendKeysOn(selector, [ RealKeys.text('tab') ]);
+  };
+
+  const setupInput = (insert: (marker: SugarElement<Node>, element: SugarElement<Node>) => void, editor: Editor) => {
+    const input = SugarElement.fromTag('input');
+    insert(TinyDom.container(editor), input);
+    return input;
+  };
+
+  const setupInputBefore = (editor: Editor) => setupInput(Insert.before, editor);
+  const setupInputAfter = (editor: Editor) => setupInput(Insert.after, editor);
+
+  const pIsFullscreen = (fullscreen: boolean) => Waiter.pTryUntilPredicate('Waiting for fullscreen mode to ' + (fullscreen ? 'start' : 'end'), () => {
+    if (fullscreen) {
+      return NativeFullscreen.getFullscreenElement(document) === document.body;
+    } else {
+      return NativeFullscreen.getFullscreenElement(document) === null;
+    }
+  });
+
+  const pToggleFullscreen = async (editor: Editor, nativeMode: boolean, fullscreen: boolean) => {
+    TinyContentActions.keystroke(editor, 121, { alt: true });
+    await FocusTools.pTryOnSelector('Assert toolbar is focused', SugarDocument.getDocument(), 'div[role=toolbar] .tox-tbtn');
+    await RealKeys.pSendKeysOn('div[role=toolbar] .tox-tbtn', [ RealKeys.text('enter') ]);
+    if (nativeMode) {
+      await pIsFullscreen(fullscreen);
+      await Waiter.pWait(500);
```

Comment:
What is this for? If we need to wait for the next execution cycle we could do `Waiter.pWait(0);`

Reply:
I was getting some weird failures with animations locally when going in and out from fullscreen mode with native 😅